### ISSUE TITLE
Consider adding a Gentoo ebuild

### ISSUE THREAD

**Issue by oakad** [AdoptOpenJDK/openjdk-build#464]
I understand, that this may be a somewhat far fetched request, but still.

Gentoo is a rather convenient distro for testing "unusual" software configurations on "usual" platforms and reigns supreme when deploying "usual" software onto "unusual" hardware platforms (custom embedded boards, etc).

This project had done a most impressive job automating the OpenJDK builds. It appears, that with relatively little additional effort, a Gentoo ebuild can also be maintained within its scope, with considerable possible benefits for the community.

Until now, Gentoo benefited from the generous support of the IcedTea project maintainer, but it appears, that IcedTea will not be unable to keep up with OpenJDK releases.

**Comment by sxa555** [comment]
@oakad Do you have the skills to put together what's required? Like you I suspect it wouldn't be too hard although I haven't used gentoo myself for a while and it'll be a while before I personally woulkd be able to spend any time on it

**Comment by oakad** [comment]
Yes, I am working on an ebuild.

The problem is, Gentoo Java community is basically defunct right now, so the ebuild needs a good home where it will be visible for the community and maintained.

Shall I offer a PR when I'm ready?

**Comment by karianna** [comment]
@oakad Yes please. Not we'll be swapping build scripts over to #320 shortly.

**Comment by nelsongraca** [comment]
@oakad Hi are you still working on this? I too would like to have ebuilds since I use Gentoo on a daily basis.

**Comment by chewi** [comment]
I am working on an official Gentoo source ebuild. It should be ready within days.

**Comment by nelsongraca** [comment]
@chewi Thank you. Any chance of supporting 8 too? From what I see it exists only for 11. 9 and 10 since they are not LTS I don't care much, but I guess someone might find them useful too.

**Comment by karianna** [comment]
Hi @chewi - thanks for your support of Adopt on this.  Are you able to connect us to the new maintainer for Java on Gentoo?

**Comment by chewi** [comment]
@karianna, as @oakad said, the team is almost defunct. The best contact going forwards is @monsieurp but I don't know how much work he is going to put into the JVM packages.

**Comment by nelsongraca** [comment]
I am willing to help on the efforts of keeping the ebuilds up to date, my knowledge is still limited but I have been keeping some ebuilds on a personal overlay in order to support some other software, so I believe that with some base to continue maintain them I can help.

**Comment by karianna** [comment]
@nelsongraca Thanks!  Are you a member of our slack org?

**Comment by nelsongraca** [comment]
I am not, but I will happily join

**Comment by oakad** [comment]
@nelsongraca My version of the ebuild is here, btw: https://github.com/gentoo/gentoo/pull/9835

**Comment by chewi** [comment]
That's the openjdk package in the official repo now.

**Comment by oakad** [comment]
@chewi Those are good news.

But if you are stepping down from the java maintenance, whom do we turn to for ongoing support in the future?

**Comment by gyakovlev** [comment]
>  Any chance of supporting 8 too? From what I see it exists only for 11. 9 and 10 since they are not LTS I don't care much, but I guess someone might find them useful too.

Adding ebuilds for 8 soon.
I'm now maintaining  openjdk related ebuilds in gentoo.

**Comment by gyakovlev** [comment]
both openjdk8 and openjdk11 are in gentoo now, both source and binary flavors, so this probably should be closed.
there are no plans to add 9 or 10, sorry, only LTS.

**Comment by stikonas** [comment]
> there are no plans to add 9 or 10, sorry, only LTS.

Does this mean there will be no way to bootstrap openjdk 11 from gcc? It was possible (although one had to enable java overlay) to bootstrap openjdk 8 from gcc (gcc6[gcj]->icedtea:7->icedtea:8->openjdk:8) and ensure that there is no Trusting Trust attack in openjdk.

**Comment by gyakovlev** [comment]
@stikonas  there is no way unless someone creates ebuilds for openjdk:9 and maybe :10 as well.
I have no plans or time to do that and will not add this to ::gentoo repo because it's effectively dead, but if someone up to doing work it can live in an overlay for users who prefer the long bootstrap road. also you'll most likely run into problems emerging old java with fresh system libs.

I think openjdk:9 should be able to bootstrap 11, but unsure.
ebuild for openjdk:9 is not that hard to create using existing ebuilds as an example, and taking bits from 8 and 11.

**Comment by stikonas** [comment]
> @stikonas there is no way unless someone creates ebuilds for openjdk:9 and maybe :10 as well.
> I have no plans or time to do that and will not add this to ::gentoo repo because it's effectively dead, but if someone up to doing work it can live in an overlay for users who prefer the long bootstrap road. also you'll most likely run into problems emerging old java with fresh system libs.
> 
> I think openjdk:9 should be able to bootstrap 11, but unsure.
> ebuild for openjdk:9 is not that hard to create using existing ebuilds as an example, and taking bits from 8 and 11.

I've created an overlay with required ebuilds:
https://git.stikonas.eu/andrius/openjdk-overlay
Maybe this would be suitable for inclusion into java-overlay?

OpenJDK 9 ebuild required a few changes, in particular I disabled treating compiler warnings as errors.

OpenJDK 10 ebuild just worked, no extra work was needed.
I've also added OpenJDK 11 ebuild that accepts building with SLOT-1.
